src/string.cr

```diff
@@ -2177,6 +2191,23 @@ class String
     last_index
   end
 
+  # ditto
+  def rindex(search : Regex, offset = 0)
+    offset += size if offset < 0
+    return nil unless 0 <= offset <= size
+
+    matchresult = nil
+    self[0, self.size - offset].scan(search) do |matchdata|
+      matchresult = matchdata
+    end
+
+    if index = matchresult.try &.begin(0)
```

Comment:
Same here, `matchresult.try &.begin` is good

Also, we prefer names like `match_result` and `match_data` instead of `matchresult` and `matchdata`, which are slightly harder to read